torch/sparse/_triton_ops.py

```diff
@@ -459,7 +462,7 @@ def _bsr_strided_dense_rowspace_kernel(
             dense_block = tl.load(dense_block_ptrs + dense_tiled_row_stride * dense_row_idx)
 
             # do block mm
-            output_acc_block += tl.dot(values_block, dense_block, allow_tf32=allow_tf32)
+            output_acc_block += tl.dot(values_block, dense_block, allow_tf32=allow_tf32, out_dtype=acc_dtype)
```

Comment:
This feels like a significant bugfix

Reply:
Line 337 above has `acc_block += tl.dot(mat1_block, mat2_block, allow_tf32=allow_tf32)` and also doesn't pass the out_dtype.

Reply:
Could the change in performance come from adjusting the out_dtype here? Maybe if so the call to contiguous isn't what's needed, but a different accumulation dtype which in turn influences which instruction is used to perform the dot.

Reply:
> This feels like a significant bugfix

Actually, it is not too bad: the default dtype for `tl.dot` output is float32 which coincides with the accumulator dtype for bfloat16/float16 inputs. So, the fix is effective only for float32 input type that uses float64 accumulator dtype.

> Line 337 above... does not specify `out_dtype`.

Fixed.

> Could the change in performance come from adjusting the out_dtype here?

No: Disabling `out_dtype` produces the same performance improvement figure as in the description.

> Maybe if so the call to contiguous isn't what's needed...

That would be nice indeed but unfortunately, it is not the case here. Notice that the performance improvement for float32 inputs is present for larger input and block sizes and I would attribute this effect to minimizing cache misses (or similar phenomena) when operating on contiguous inputs (although, the cache misses are not directly measurable here). For smaller input sizes, cache misses are less likely and the overhead of the `contiguous()` call becomes apparent in a slight reduction of the performance.

Reply:
In fact, considering
https://github.com/openai/triton/blob/36859aebff1897f3bdf6dd19ba775e5961848943/python/triton/language/semantic.py#L1294C1-L1305C34
`out_dtype` value is in many cases just ignored by triton and specifying `out_dtype` for `float32` inputs has no effect.